### DISCUSSION TITLE
General Questions about NXT

### DISCUSSION THREAD

**Discussion by marcovidaurre** [nxt-dev/nxt_editor#273] [Q&A]
Hi ,

I'm very new to NXT and had a couple of questions.  

Is it possible to dynamically generate nodes in NXT and if so how ? 

Also what is the best way to generate attributes on my nodes from within the code block ?  

Thanks

**Comment by MichaelAldrich** [comment]
> Is it possible to dynamically generate nodes in NXT and if so how ?

"Possible", yes. There's no designed workflow or api except the editor, so doing it in python is a search and find mission.

What kind of graphs are you planning to generate? Maybe there's an easier way to give you what you want. The save files are just json after all.

**Reply by MichaelAldrich** [reply]
Oh, I didn't notice you said "within the code block", without reading the code my memory tells me that graphs do not have access to the editor api(intentionally) so that would be an even deeper, but still possible hack, for grabbing the main window as described here: https://github.com/nxt-dev/nxt_editor/discussions/261#discussioncomment-5732110

**Reply by marcovidaurre** [reply]
Hi ,

Thanks so much for your quick reply.  I'll give you a simple example of what I'm trying to do.

Lets say I want to iterate through all of the shots in Maya's sequence manager and send them to the farm for hardware render.

The shots and camera names may be different from scene to scene .  I'd like a node to represent every shot object in the sequencer so multiple nodes. Then I would like to write out two or three attributes from the shot object into each node object. 

That means I need to generate my graph on the fly and populate those attributes on each node. 

Would I just do that from an external script that has access to all of NXT apis?  

Hope that makes sense
